oeps/oep-0044-bp-changelogs.rst

```diff
@@ -0,0 +1,285 @@
+* If you decide that reverting will cause more harm for some reason, and you do not intend to revert, please let this be a part of your communication.
+
+The reason it is important that only MAJOR versions include backward incompatible change is because when people upgrade dependencies, some will only read changelogs for MAJOR versions because of the potential backward incompatibilities, and may not read notes for MINOR or PATCH upgrades.
+
+Frontend Libraries
+~~~~~~~~~~~~~~~~~~
+
+Like all Open edX repositories, frontend libraries should also include a CHANGELOG.rst. However, as you'll see below, this file will simply point to the GitHub Releases. Here is an example CHANGELOG.rst::
+
+  Changelog
+  =========
+
+  All notable changes to this project will be documented as `GitHub Releases`_.
+
+  This project adheres to `Semantic Versioning`_.
+
+  .. _GitHub Releases: https://github.com/edx/frontend-build/releases
+  .. _Semantic Versioning: https://semver.org/spec/v2.0.0.html
+
+
+All new frontend libraries have been using an npm package called `semantic-release`_.
+
+The `semantic-release`_ package automates the following:
+
+* Incrementing the version appropriately for each PR merge, based on a well formatted commit comment. See the `semantic-release`_ README for details. A pre-commit hook is typically run to ensure the commit message is appropriately formatted for `semantic-release`_.
+
+  * One important formatting detail is to include "BREAKING CHANGE" at the beginning of a line in the commit description to get a new major version.
+
+* Publishing the changelog message as a GitHub Release in the repository, using the commit message. Here is an example `GitHub Release changelog for edx/frontend-build`_.
+
+As noted above, please update the changelogs as necessary to add clarifications or customize the messaging for the audience of the changelog, especially because these messages originally come from commit messages, which may have been written for a different audience.
+
+.. _semantic-release: https://github.com/semantic-release/semantic-release
+.. _GitHub Release changelog for edx/frontend-build: https://github.com/edx/frontend-build/releases
+
+Python Libraries
+~~~~~~~~~~~~~~~~
+
+Like all Open edX repositories, Python libraries should also include a CHANGELOG.rst. This file may also simply point to the GitHub Releases. Here is an example CHANGELOG.rst::
+
+  Changelog
+  =========
+
+  All notable changes to this project will be documented as `GitHub Releases`_.
+
+  This project adheres to `Semantic Versioning`_.
+
+  .. _GitHub Releases: https://github.com/edx/frontend-build/releases
+  .. _Semantic Versioning: https://semver.org/spec/v2.0.0.html
+
+Many Python libraries publish packages to PyPI when a version is tagged. Publishing a GitHub Release both tags the version and allows one to write an appropriate changelog message. This is the recommended process at this time.
```

Comment:
Do we have a recommendation for how to track the work-in-progress until the release, or are we assuming for a Python library that every change will result in publishing a new version?

Reply:
Mostly we publish every change.

Reply:
This defeats the entire purpose of a changelog. If I want to read the git history of a repo I run `git log`. There are too many commits that do not belong in a changelog. A changelog is primarily for users of a tool.

Reply:
What you are calling "too many commits" is simply a result of our current process.  For our libraries, we release a new version of the library as soon as we need a new feature or fix.  We do not slowly accumulate changes until we are ready for some grand release.  So, in practice, there is typically one PR per version/release.
I think we would both agree that users of the library need to know what is contained in each version. So, maybe you are asking for a different process with fewer versions?

Reply:
I don't think there are too many commits in the repository, and I think it's great that we release a new version for every fix and feature. See my comment above for how I think new versions should be described.